src/osemgrep/networking/Scan_helper.ml

```diff
@@ -115,3 +115,106 @@ let report_failure ~dry_run ~token ~scan_id exit_code =
     | Error (code, msg) ->
         Error
           ("API server returned " ^ string_of_int code ^ ", this error: " ^ msg)
+
+(* TODO the server reply would be specified in ATD *)
+let extract_errors data =
+  try
+    match JSON.json_of_string data with
+    | JSON.Object xs -> (
+        match List.assoc_opt "errors" xs with
+        | Some (JSON.Array errs) ->
+            List.iter
+              (fun err ->
+                match err with
+                | JSON.Object xs -> (
+                    match List.assoc_opt "message" xs with
+                    | Some (String s) ->
+                        Logs.warn (fun m ->
+                            m "Server returned following warning: %s" s)
+                    | _else ->
+                        Logs.err (fun m ->
+                            m "Couldn't find message in %s"
+                              (JSON.string_of_json err)))
+                | _else ->
+                    Logs.err (fun m ->
+                        m "Couldn't find message in %s"
+                          (JSON.string_of_json err)))
+              errs
+        | _else ->
+            Logs.err (fun m ->
+                m "Couldn't find errors in %s"
+                  (JSON.string_of_json (JSON.Object xs))))
+    | json ->
+        Logs.err (fun m -> m "Not a json object %s" (JSON.string_of_json json))
+  with
+  | e ->
+      Logs.err (fun m ->
+          m "Failed to decode server reply as json %s: %s"
+            (Printexc.to_string e) data)
+
+(* TODO the server reply should be specified in ATD *)
```

Comment:
same here

Reply:
done in [d8aaaf4](https://github.com/returntocorp/semgrep/pull/8168/commits/d8aaaf41a50b5dc72a71fb776e75a8422f4b87d5)